src/Build.UnitTests/Evaluation/Expander_Tests.cs

```diff
@@ -4978,6 +4978,56 @@ public void ExpandItem_ConvertToStringUsingInvariantCultureForNumberData_Respect
             }
         }
 
+        [Theory]
+        [InlineData("getType")]
+        [InlineData("GetType")]
+        [InlineData("gettype")]
+        public void GetTypeMethod_ShouldNotBeAllowed(string methodName)
+        {
+            using (var env = TestEnvironment.Create())
+            {
+                var root = env.CreateFolder();
+
+                var projectFile = env.CreateFile(root, ".proj",
+                    @$"<Project>
+    <PropertyGroup>
+        <foo>aa</foo>
+        <typeval>$(foo.{methodName}().FullName)</typeval>
+    </PropertyGroup>
+</Project>");
+                var exception = Should.Throw<InvalidProjectFileException>(() =>
+                {
+                    new ProjectInstance(projectFile.Path);
+                });
+                exception.BaseMessage.ShouldContain($"The function \"{methodName}\" on type \"System.String\" is not available for execution as an MSBuild property function.");
```

Comment:
nit: Wouldn't this assertion fail with non-English locale? Would it be enough to look for `MSB4185`?

Reply:
Thank you for pointing out. 
In my opinion verifying that code exists in the message does not provide 100% certainty that the message is correct , + it has additinal protection on message error change 
what do you think if before the test to set the currentCultureInfo?  
```
currentThread.CurrentCulture = new CultureInfo("en");
currentThread.CurrentUICulture = new CultureInfo("en");
```

Reply:
Thank you for the quick response. I don't have a strong opinion. Setting the culture (and restoring it after the test) looks like a good approach. Or maybe asserting that the code and `methodName` are both present would also provide reasonable coverage.